lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -0,0 +1,228 @@
+      end
+
+      private
+
+      def add_customer_data(post, options)
+        post.merge!(options.slice(*CUSTOMER_DATA))
+      end
+
+      def add_address(post, options)
+        return unless post[:card] && post[:card].kind_of?(Hash)
+        if address = options[:billing_address] || options[:address]
+          post[:card][:billingAddress] = {}
+          post[:card][:billingAddress][:street] = address[:address1] if address[:address1]
+          post[:card][:billingAddress][:houseNumberOrName] = address[:address2] if address[:address2]
+          post[:card][:billingAddress][:postalCode] = address[:zip] if address[:zip]
+          post[:card][:billingAddress][:city] = address[:city] if address[:city]
+          post[:card][:billingAddress][:stateOrProvince] = address[:state] if address[:state]
+          post[:card][:billingAddress][:country] = address[:country] if address[:country]
+        end
+      end
+
+      def add_invoice(post, money, options)
+        amount = {
+          value: amount(money),
+          currency: options[:currency] || currency(money)
+        }
+        post[:reference] = options[:reference]
+        post[:amount] = amount
+      end
+
+      def add_invoice_for_modification(post, money, authorization, options)
+        amount = {
+          value: amount(money),
+          currency: options[:currency] || currency(money)
+        }
+        post[:modificationAmount] = amount
+      end
+
+      def add_payment(post, payment)
+        card = {
+          expiryMonth: payment.month,
+          expiryYear: payment.year,
+          holderName: payment.name,
+          number: payment.number,
+          cvc: payment.verification_value
+        }
+        card.delete_if{|k,v| v.blank? }
+        requires!(card, :expiryMonth, :expiryYear, :holderName, :number, :cvc)
+        post[:card] = card
+      end
+
+      def add_references(post, authorization, options = {})
+        post[:originalReference] = authorization
+        post[:reference] = options[:reference]
+      end
+
+      def parse(body)
+        return {} if body.blank?
+        JSON.parse(body)
+      end
+
+      def commit(action, parameters)
+        url = (test? ? test_url : live_url)
+
+        begin
+          raw_response = ssl_post("#{url}/#{action.to_s}", post_data(action, parameters), request_headers)
+          response = parse(raw_response)
+        rescue ResponseError => e
+          raw_response = e.response.body
+          response = parse(raw_response)
+        end
+
+        success = success_from(action, response)
+        Response.new(
+          success,
+          message_from(action, response),
+          response,
+          authorization: authorization_from(response),
+          test: test?,
+          error_code: success ? nil : error_code_from(response)
+        )
+
+      end
+
+      def basic_auth
+        Base64.encode64("#{@username}:#{@password}").strip.delete("\r\n")
```

Comment:
I believe if you use `strict_encode64`, you won't need to manually strip off the newline info.